src/main.rs

```diff
@@ -53,7 +62,11 @@ fn main() -> Result {
     // configure and start web server
     let conf = read_config()?;
     let context = Context {
-        identity_store: Box::new(Ldap::new(conf.ldap)),
+        identity_store: match conf.main.identity_store.as_str() {
+            "ldap" => Box::new(Ldap::new(conf.ldap)),
+            "json" => Box::new(Json::new(conf.json_identity)),
+            &_ => {panic!("Unknown identity server.");}
```

Comment:
You don't need to take `_` by reference. And don't panic, instead return an error (You don't need to do this, if using an enum instead of a string).